tests/dsl_unittest.py

```diff
@@ -2023,17 +2024,31 @@ async def expect_pass(self):
                     ).for_call().to_return_value("mocked").and_assert_called_once()
                     assert await SomeClass.do_something() == "mocked"
 
-        examples = {}
-
-        for all_top_level_context in Context.all_top_level_contexts:
-            for example in all_top_level_context.all_examples:
-                examples[example.name] = example
+        examples = _get_name_to_examples()
 
         self.run_example(examples["expect pass"])
 
         with self.assertRaisesRegex(AssertionError, "calls did not match assertion"):
             self.run_example(examples["expect fail"])
 
+    def test_leaked_tasks_fails_test_run(self):
+        async def dummy_async_func():
+            pass
+
+        @context
+        def fail_top(context):
+            @context.example
+            async def spawn_task_but_dont_await(self):
+                # once we stop supporting python 3.6 we can use
```

Comment:
Can we have an `if python version <= 3.6` here please? It helps cleanup in the future, as we can merely delete code, instead of adding.